llpc/lower/llpcSpirvLowerCfgMerges.cpp

```diff
@@ -0,0 +1,648 @@
+ */
+#include "llpcSpirvLowerCfgMerges.h"
+#include "SPIRVInternal.h"
+#include "llpcContext.h"
+#include "llpcDebug.h"
+#include "llpcSpirvLower.h"
+#include "llpcSpirvLowerUtil.h"
+#include "lgc/Builder.h"
+#include "llvm/ADT/DepthFirstIterator.h"
+#include "llvm/ADT/SetVector.h"
+#include "llvm/ADT/SmallSet.h"
+#include "llvm/IR/InlineAsm.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/Pass.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/raw_ostream.h"
+
+#define DEBUG_TYPE "llpc-spirv-lower-cfg-merges"
+
+// Silence cppcheck for LLVM_DEBUG by defining macro.
+// Note: cppcheck-suppress does not reliably work for this.
+#ifndef LLVM_DEBUG
+#define LLVM_DEBUG(...)
+#endif
+
+using namespace llvm;
+using namespace SPIRV;
+using namespace Llpc;
+
+// -enable-loop-reconvergence: force enable loop reconvergence transform
+static cl::opt<bool> EnableLoopReconvergence("enable-loop-reconvergence",
+                                             cl::desc("Force enable loop reconvergence transform"), cl::init(false));
+
+namespace Llpc {
+
+// =====================================================================================================================
+// Initializes static members.
+char LegacySpirvLowerCfgMerges::ID = 0;
+
+// =====================================================================================================================
+// Pass creator, creates the pass of SPIR-V lowering terminator operations
+ModulePass *createLegacySpirvLowerCfgMerges() {
+  return new LegacySpirvLowerCfgMerges();
+}
+
+// =====================================================================================================================
+LegacySpirvLowerCfgMerges::LegacySpirvLowerCfgMerges() : ModulePass(ID) {
+}
+
+// =====================================================================================================================
+// Executes this SPIR-V lowering pass on the specified LLVM module.
+//
+// @param [in/out] module : LLVM module to be run on (empty on entry)
+bool LegacySpirvLowerCfgMerges::runOnModule(Module &module) {
+  return Impl.runImpl(module);
+}
+
+// =====================================================================================================================
+// Executes this SPIR-V lowering pass on the specified LLVM module.
+//
+// @param [in/out] module : LLVM module to be run on (empty on entry)
+// @param [in/out] analysisManager : Analysis manager to use for this transformation
+PreservedAnalyses SpirvLowerCfgMerges::run(Module &module, ModuleAnalysisManager &analysisManager) {
+  bool changed = runImpl(module);
+  // In practice there are unlikely to be any analyses this early, but report accurate status anyway.
+  return changed ? PreservedAnalyses::none() : PreservedAnalyses::all();
+}
+
+class OpPrinter {
+  BasicBlock *m_bb;
+  Function *m_func;
+
+public:
+  OpPrinter(BasicBlock *BB) : m_bb(BB), m_func(nullptr) {}
```

Comment:
Maybe rename `BB` to `block`. There are several such places.

Reply:
Sure BB -> block, although BB is very common LLVM notation for a basic block reference.